### PR TITLE
Artist list

### PR DESCRIPTION
**Pull request by ninapeters**
Please review and leave comments. c:

### CODE REVIEW

File: src/components/ArtistList.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  span {
+    display: block;
+    font-size: 75%;
+    font-weight: 300;
+    padding-bottom: 6px;
+    text-transform: uppercase;
+  }
+
+  background-color: var(--primary);
+  color: var(--font);
```
Comment (mpagels) [inline src/components/ArtistList.js:41]: The variable name is imho not really meaningful

File: src/services/getTrackData.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function getTrackData() {
+  return fetch('http://localhost:4000/tracks').then((res) => res.json())
```
Comment (mpagels) [inline src/services/getTrackData.js:2]: I would recommend some error catching if server is not available.

**Review by mpagels — CHANGES_REQUESTED** [review]
Very good job. Keep up! Found only one or two little things and one big issue.
Please review.

File: src/components/ArtistList.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import styled from 'styled-components/macro'
+import PropTypes from 'prop-types'
+
+export default ArtistList
+
+ArtistList.propTypes = {
+  artists: PropTypes.array,
```
Comment (felix-horn) [inline src/components/ArtistList.js:7]: Is this PropType required? If so, then it should be: 

`artists: PropTypes.array.isRequired,`

Comment (felix-horn) [reply src/components/ArtistList.js:7]: Then again, in line 14 it seems a potential error / missing array is handled. So the above comment is probably obsolete.

File: src/components/ArtistList.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+function ArtistList({ artists }) {
+  return (
+    <div>
+      <ListStyled>
+        {artists?.map(({ artist, songTitle, id }) => (
```
Comment (felix-horn) [inline src/components/ArtistList.js:14]: What exactly does the `?` do? 
My inerpretation would be: If `artists` exist, then `artist.map(...`, however, if `artists` does not exist, then don't map anything. Is that correct?

File: src/components/ArtistList.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+          <ListItemStyled key={id}>
+            <span>{artist}</span>
+            {songTitle}
```
Comment (felix-horn) [inline src/components/ArtistList.js:17]: Why did you enclose `{artist}` in a span but `{songTitle}` does not have a tag?

File: src/components/ArtistList.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+export default ArtistList
+
+ArtistList.propTypes = {
+  artists: PropTypes.array,
+}
+
+function ArtistList({ artists }) {
+  return (
+    <div>
```
Comment (felix-horn) [inline src/components/ArtistList.js:12]: What's the all enclosing `<div>` used for? I believe it's obsolete as <ListStyled> is only one element.

File: src/components/ArtistList.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    font-weight: 300;
+    padding-bottom: 6px;
+    text-transform: uppercase;
+  }
+
+  background-color: var(--primary);
+  color: var(--font);
+  font-size: 1em;
+  font-weight: 600;
+  padding: 20px;
```
Comment (felix-horn) [inline src/components/ArtistList.js:44]: I would have expected this block to right after `const ListItemStyled...` - but I think it doesn't matter?

File: src/components/ArtistList.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+```js
+<ArtistList
+  artists={[
+    {
+      artist: 'Dance Gavin Dance',
+      songTitle: 'We Own The Night',
+      id: 'jkas872301jklas',
```
Comment (felix-horn) [inline src/components/ArtistList.md:7]: I think, you don't have to hand in the id at for the Styleguidist, but I'm not too sure.. :)

File: src/components/ArtistList.test.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('ArtistList', () => {
+  it('renders correctly', () => {
+    const { container } = render(
+      <ArtistList
+        artists={[
+          {
+            artist: 'Dance Gavin Dance',
+            songTitle: 'We Own The Night',
+            id: 'jkas872301jklas',
```
Comment (felix-horn) [inline src/components/ArtistList.test.js:12]: Same here, I think the id is obsolete here, as it's not rendered.

File: src/index.js [hunk 10]
```diff
@@ -1,10 +1,12 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
 import App from './App'
+import GlobalStyle from './styles/GlobalStyle'
 import reportWebVitals from './reportWebVitals'
```
Comment (felix-horn) [inline src/index.js:5]: I deleted this line.

File: src/App.js [hunk 11]
```diff
@@ -1,9 +1,22 @@
...
+import normalizeArtists from './services/normalizeArtists'
+
+export default App
+
 function App() {
+  const [artistData, setArtistData] = useState([])
+
+  useEffect(() => {
+    getTrackData().then((tracks) =>
+      setArtistData(normalizeArtists(tracks.items))
```
Comment (felix-horn) [inline src/App.js:13]: Uuuuuhhhh - took me a while to get the `normalizeArtists-fn`, but I like it!

**Review by felix-horn — COMMENTED** [review]
Looks great! You probably have a separate project for the server you wrote, right? Does that also need to be reviewed?

**Review by Hecktors — COMMENTED** [review]
Love your music app!

File: src/components/ArtistList.js (hunk 1 above)
Comment (ninapeters) [reply src/components/ArtistList.js:41]: Good point, I'll change that.

File: src/services/getTrackData.js (hunk 2 above)
Comment (ninapeters) [reply src/services/getTrackData.js:2]: > Looks great! You probably have a separate project for the server you wrote, right? Does that also need to be reviewed?

I have a json server installed in my react app. You can find the data in "src / data / db.json"

File: src/components/ArtistList.js (hunk 4 above)
Comment (ninapeters) [reply src/components/ArtistList.js:14]: If `artists` is `null` or `undefined`, the map is not executed and there is no error message: `Optional Chaining`

File: src/components/ArtistList.js (hunk 5 above)
Comment (ninapeters) [reply src/components/ArtistList.js:17]: `Artist` and `SongTitle` are both enclosed in the `list tag`. The `artist` gets an extra styling, which is why it is included in a `span tag`.

File: src/components/ArtistList.md (hunk 8 above)
Comment (ninapeters) [reply src/components/ArtistList.md:7]: So it's complete for me, even if it's not rendered.

File: src/styles/GlobalStyle.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import { createGlobalStyle } from 'styled-components/macro'
+
+export default createGlobalStyle`
+* {
+    box-sizing: border-box;
+}
+
+body {
+    font-family: sans-serif;
```
Comment (dajanaheinze) [inline src/styles/GlobalStyle.js:9]: check css sorting isn't diffrent to componend sorting

File: src/App.js [hunk 13]
```diff
@@ -1,9 +1,20 @@
...
+import ArtistList from './components/ArtistList'
+
+export default App
+
 function App() {
+  const [artistData, setArtistData] = useState([])
+
+  useEffect(() => {
+    setArtistData(normalizeArtists(data.tracks.items))
+  }, [])
```
Comment (mmurakaru) [inline src/App.js:13]: Consider puting line 8-13 into a costum hook and return -> const {artistData} = useArtists()

File: styleguide.config.js [hunk 14]
```diff
@@ -1,6 +1,11 @@
+const path = require('path')
+
 module.exports = {
-  ignore: ['**/App.js'],
+  ignore: ['**/App.js', '**/*.test.js', '**/styles/GlobalStyle.js'],
   defaultExample: true,
```
Comment (mmurakaru) [inline styleguide.config.js:5]: Consider setting this to false since you added dummy data to your ArtistList.md 🙈

**Review by mmurakaru — CHANGES_REQUESTED** [review]
Consider mini-mini optimizations!

File: src/App.js (hunk 13 above)
Comment (ninapeters) [reply src/App.js:13]: The state is not very complex at this point. For now, I find it clearer to have it directly in the app. I'll keep it in mind for later. Thank you!

**Pull request merged**